Source/WebInspectorUI/UserInterface/Views/TreeOutline.js

```diff
@@ -402,12 +402,21 @@ WI.TreeOutline = class TreeOutline extends WI.Object
 
     _forgetTreeElementAndDescendants(element)
     {
+        if (element === this.selectedTreeElement) {
+            const suppressNotification = true;
+            element.deselect(suppressNotification);
+        }
+
         element.treeOutline = null;
         this._knownTreeElements.delete(element.identifier, element);
 
         const skipUnrevealed = false;
         const dontPopulate = true;
         for (let current = element.children[0]; current; current = current.traverseNextTreeElement(skipUnrevealed, element, dontPopulate)) {
+            if (current === this.selectedTreeElement) {
+                const suppressNotification = true;
+                current.deselect(suppressNotification);
+            }
```

Comment:
i think this might already be handled in `WI.TreeOutline.prototype.removeChildAtIndex` (but only if `!suppressSelectSibling`) so maybe it just needs to happen in `WI.TreeOutline.prototype.removeChildren`?  or perhaps instead of calling `WI.TreeElement.prototype.deselect` we just need to call `this.selectedTreeElement = null;` once at the very end?